src/Http/Routing/test/testassets/RoutingWebSite/UseEndpointRoutingStartup.cs

```diff
@@ -173,7 +173,7 @@ private void SetupBranch(IApplicationBuilder app, string name)
             app.UseRouting();
             app.UseEndpoints(endpoints =>
             {
-                endpoints.MapGet("api/get/{id}", (context) => context.Response.WriteAsync($"{name} - API Get {context.GetRouteData().Values["id"]}"));
+                endpoints.MapGet("api/get/{id}", (context) => context.Response.WriteAsync($"{name} - API Get {context.Request.RouteValues["id"]}"));
```

Comment:
@rynowak RouteData is null now.

Reply:
It should be created lazily if GetRouteData is called for backwards compat

Reply:
Maybe, I added a change that news it up over the current route values. We'll need to decide what to do here all up.

Reply:
This is fixed but I'm going to leave the 3.0 way in here